src/Raven.Server/Documents/Revisions/RevisionsStorage.cs

```diff
@@ -2477,8 +2510,69 @@ public IEnumerable<Document> GetRevisionsBinEntries(DocumentsOperationContext co
             }
         }
 
+        public List<(string Id, long Etag)> GetDeletedRevisionsIds(DocumentsOperationContext context, DateTime before,
+            long batchSize, ref long lastEtag, CancellationToken token)
+        {
+            var ids = new List<(string Id, long Etag)>();
+            var count = 0L;
+            var table = new Table(RevisionsSchema, context.Transaction.InnerTransaction);
+            using var _ = GetEtagAsSlice(context, lastEtag, out var startSlice);
+
+            foreach (var result in table.SeekForwardFrom(RevisionsSchema.Indexes[DeleteRevisionEtagSlice], startSlice, 0))
+            {
+                ref var reader = ref result.Result.Reader;
+                var deleteRevision = TableValueToRevision(context, ref reader, DocumentFields.Id);
+                var deletedEtag = TableValueToEtag((int)RevisionsTable.DeletedEtag, ref reader);
+                lastEtag = deletedEtag;
+                if (deleteRevision.LastModified >= before)
+                    break;
```

Comment:
we break here, but already changed the `lastEtag`, isn't this a problem, since we might skip this next time?
Would it be safe to move `lastEtag = deletedEtag;` to the next line?

Reply:
It not right to do this,
because in the next time we'll enter here we'll start from etag that doesn't exist..